community/server/src/main/java/org/neo4j/server/rest/web/CypherService.java

```diff
@@ -118,6 +130,11 @@ public Response cypher(String body,
                 includePlan = result.getQueryExecutionType().requestedExecutionPlanDescription();
             }
 
+            if ( periodicCommitQuery )
+            {
+                handler.setTransaction( database.beginTx() );
```

Comment:
Shouldn't we begin this transaction with the same type and AccessMode as the transaction we are closing above?